### DISCUSSION TITLE
sync: Listing failed, stop syncing

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2849] [General]
**What happened**:

```
juicefs sync s3://bucket1 s3://bucket2 --update --delete-dst
...
<WARNING>: Fail to list: found invalid key jfs/chunks/0/135/135272_0_4245 from List, prefix: , marker: jfs/chunks/0/179/179663_0_4194304, retry again [sync.go:156]
<ERROR>: Fail to list after jfs/chunks/0/179/179663_0_4194304: found invalid key jfs/chunks/0/135/135272_0_4245 from List, prefix: , marker: jfs/chunks/0/179/179663_0_4194304 [sync.go:166]
<ERROR>: Listing failed, stop syncing, waiting for pending ones [sync.go:667]
```

**Environment**:
- JuiceFS version: 1.0.0

**Comment by davies** [comment]
What's the S3-compatible storage used here? The error means List returned expected key (less than marker).

**Comment by solracsf** [comment]
[Idrive-e2](https://www.idrive.com/e2/faq#s3-api), tested successfully trough `objbench` (all pass but the latest 3).
Please note that other buckets complete successfully, only this one is having this problem.

**Comment by davies** [comment]
The returned key `jfs/chunks/0/135/135272_0_4245` is less than marker `jfs/chunks/0/135/135272_0_4245 from List, prefix: , marker: jfs/chunks/0/179/179663_0_4194304` , which break the rule that returned keys should be greater than marker.

```
	// Marker is where you want Amazon S3 to start listing from. Amazon S3 starts
	// listing after this specified key. Marker can be any key in the bucket.
	Marker *string `location:"querystring" locationName:"marker" type:"string"`
```

I think there is nothing we can do here. Please report this to Idrive-e2.

**Reply by solracsf** [reply]
Fixed by the object storage provider, thanks.
